Swinject/Assembler.swift

```diff
@@ -0,0 +1,99 @@
+//
+//  Assembler.swift
+//  Swinject
+//
+//  Created by mike.owens on 12/9/15.
+//  Copyright © 2015 Swinject Contributors. All rights reserved.
+//
+
+
+/// The `Assembler` provides a means to build a container via `AssemblyType` instances.
+public class Assembler {
+    
+    /// the container that each assembly will build its `Service` definitions into
+    private let container: Container
+    
+    /// expose the container as a resolver so `Service` registration only happens within an assembly
+    public var resolver: ResolverType {
+        return container
+    }
+    
+    /// Will create an empty `Assembler`
+    ///
+    /// - parameter container: the baseline container
+    ///
+    public init(container: Container? = Container()) {
+        self.container = container!
+    }
+    
+    /// Will create a new `Assembler` with the given `AssemblyType` instances to build a `Container`
+    ///
+    /// - parameter assemblies:         the list of assemblies to build the container from
+    /// - parameter propertyLoaders:    a list of property loaders to apply to the container
+    /// - parameter container:          the baseline container
+    ///
+    public init(assemblies: [AssemblyType], propertyLoaders: [PropertyLoaderType]? = nil, container: Container? = Container()) throws {
+        self.container = container!
+        if let propertyLoaders = propertyLoaders {
+            for propertyLoader in propertyLoaders {
+                try self.container.applyPropertyLoader(propertyLoader)
+            }
+        }
+        runAssemblies(assemblies)
+    }
+    
+    /// Will apply the assembly to the container. This is useful if you want to lazy load an assembly into the assembler's
+    /// container.
+    ///
+    /// If this assembly type is load aware, the loaded hook will be invoked right after the container has assembled
+    /// since after each call to `addAssembly` the container is fully loaded in its current state. If you wish to
+    /// lazy load several assemblies that have interdependencies between each other use `appyAssemblies`
+    ///
+    /// - parameter assembly: the assembly to apply to the container
+    ///
+    public func applyAssembly(assembly: AssemblyType) {
+        assembly.assemble(container)
+        if let assemblyLoadAware = assembly as? AssemblyLoadAwareType {
+            assemblyLoadAware.loaded(resolver)
+        }
+    }
```

Comment:
I noticed the case `assemblyLoadAware.loaded(resolver)` is called is not covered by any test.

![screenshot](https://cloud.githubusercontent.com/assets/965994/11718065/7ba1d3c8-9f97-11e5-87a3-daa833d0e1a1.png)

Maybe this method can be implemented as below to simplify?

    public func applyAssembly(assembly: AssemblyType) {
        applyAssemblies([assembly])
    }